### DISCUSSION TITLE
Is there any plan to support HTTP2 as transport?

### DISCUSSION THREAD

**Discussion by masokan** [nats-io/nats-server#2229] [General]
I am using gRPC in a project and exploring the possibility of using nats.io in another.  Is there any plan to support HTTP2 as transport in the future?  Thanks.

**Comment by derekcollison** [comment]
We have been tracking QUIC3 and may develop something there. HTTP2/3 do not make as much sense for NATS.io imo..

**Reply by derekcollison** [reply]
We may do something here for last mile to the clients.
Routes inside a cluster most likely not, but possible as well for Gateways.
